### ISSUE TITLE
UI is blocking when getting the "session timed out" error

### ISSUE THREAD

**Issue by wind15973** [aws-amplify/amplify-ui-swift-liveness#61]
Hello
We built an app with Amplify 2.18.0 and AmplifyUILiveness 1.1.1.
**It worked well if we followed the hints to finish the flow.**

When we simulated some failure cases like time-out, we started the face check flow and didn't finish it.
The UI was blocking for about 5 - 10 seconds when getting the first error.
The console log showed the error but there was no "TTT test" string in the first time.
After that, we got the next error. 
The console log showed the "TTT test" string and the error.
It is weird that the UI is blocking when getting the "session timed out" error.

Please help with this case.
Thank you!

**Sample code we used:**
```
FaceLivenessDetectorView(sessionID: viewModel.sessionID, region: "us-east-1", disableStartView: true, isPresented: $isPresentingLiveness) { result in
                switch result {
                case .success:
                    let _ = debugPrint("TTT success")
                case .failure(let error):
                    DispatchQueue.main.async {
                        let _ = debugPrint("TTT test")
                    }
                    let detectError = error as FaceLivenessDetectionError
                    let _ = debugPrint("TTT \(String(describing: detectError))")
                }
            }
```

**Console logs:**
<img width="1075" alt="截圖 2023-10-12 下午5 18 09" src="https://github.com/aws-amplify/amplify-ui-swift-liveness/assets/4263011/1bc7aa9e-da88-423a-add2-0ef047b549ff">

**Comment by phantumcode** [comment]
@wind15973 Thanks for submitting the issue, we'll try to reproduce the issue and investigate further.

**Comment by wind15973** [comment]
Hi @phantumcode, I created a PR to fix this issue #64. I will also try the latest release. Thanks for your help.

**Comment by phantumcode** [comment]
@wind15973 Can you provide confirmation that the latest release fixes your issue?

**Comment by wind15973** [comment]
I will test it in these two days. Will back to you soon. Thank you

**Comment by phantumcode** [comment]
@wind15973 Fix should be in the latest release [1.1.4](https://github.com/aws-amplify/amplify-ui-swift-liveness/releases/tag/1.1.4)
Your fix in #64 does not resolve the root cause.  In the original issue submitted, the error message was delayed 5-10 seconds because the error was being set on a background thread (this was fixed in the previous release).  The error messages are getting set twice because the timeout handler was getting call even after the timeout error was already set as the current state, this should be fixed in the latest release 1.1.4.

**Comment by wind15973** [comment]
Hi @phantumcode, this issue still has not been fixed in the latest release 1.2.1.
